db/org_members_groups.go

```diff
@@ -0,0 +1,273 @@
+func (ms *MongoStorage) OrganizationMemberGroups(orgAddress string, page, pageSize int) (int, []*OrganizationMemberGroup, error) {
+	// create a context with a timeout
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	filter := bson.M{"orgAddress": orgAddress}
+
+	// Count total documents
+	totalCount, err := ms.orgMemberGroups.CountDocuments(ctx, filter)
+	if err != nil {
+		return 0, nil, err
+	}
+	totalPages := int(math.Ceil(float64(totalCount) / float64(pageSize)))
+
+	// Calculate skip value based on page and pageSize
+	skip := (page - 1) * pageSize
+
+	// Set up options for pagination
+	findOptions := options.Find().
+		SetSkip(int64(skip)).
+		SetLimit(int64(pageSize))
+
+	// find the organization in the database
+	cursor, err := ms.orgMemberGroups.Find(ctx, filter, findOptions)
+	if err != nil {
+		return 0, nil, err
+	}
+	defer func() {
+		if err := cursor.Close(ctx); err != nil {
+			log.Warnw("error closing cursor", "error", err)
+		}
+	}()
+
+	var groups []*OrganizationMemberGroup
+	for cursor.Next(ctx) {
+		var group OrganizationMemberGroup
+		if err := cursor.Decode(&group); err != nil {
+			return 0, nil, err
+		}
+		groups = append(groups, &group)
+	}
+
+	if err := cursor.Err(); err != nil {
+		return 0, nil, err
+	}
+
+	return totalPages, groups, nil
+}
+
+// CreateOrganizationGroup Creates an organization member group
+func (ms *MongoStorage) CreateOrganizationMembersGroup(group *OrganizationMemberGroup) (string, error) {
+	// create a context with a timeout
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	if group == nil || group.OrgAddress == "" || len(group.MemberIDs) == 0 {
+		return "", ErrInvalidData
+	}
+
+	// check that the organization exists
+	if _, err := ms.fetchOrganizationFromDB(ctx, group.OrgAddress); err != nil {
+		if err == ErrNotFound {
+			return "", ErrInvalidData
+		}
+		return "", fmt.Errorf("organization not found: %w", err)
+	}
+	// check that the members are valid
+	err := ms.validateOrgMembers(ctx, group.OrgAddress, group.MemberIDs)
+	if err != nil {
+		return "", err
+	}
+	// create the group id
+	group.ID = primitive.NewObjectID()
+	group.CreatedAt = time.Now()
+	group.UpdatedAt = time.Now()
+	group.CensusIDs = make([]string, 0)
+
+	// Only lock the mutex during the actual database operations
+	ms.keysLock.Lock()
+	defer ms.keysLock.Unlock()
+
+	// insert the group into the database
+	if _, err := ms.orgMemberGroups.InsertOne(ctx, *group); err != nil {
+		return "", fmt.Errorf("could not create organization members group: %w", err)
+	}
+	return group.ID.Hex(), nil
+}
+
+// UpdateOrganizationMembersGroup updates an organization members group by adding
+// and/or removing members. If a member exists in both lists, it will be removed
+// TODO allow to update the rest of the fields as well. Maybe a different function?
+func (ms *MongoStorage) UpdateOrganizationMembersGroup(
```

Comment:
looking at the other method names, this should be UpdateOrganizationMemberGroup for consistency